src/db.rs

```diff
@@ -416,26 +416,26 @@ impl DB {
 
     pub fn iterator<'a: 'b, 'b>(&'a self, mode: IteratorMode) -> DBIterator<'b> {
         let readopts = ReadOptions::default();
-        self.iterator_opt(mode, &readopts)
+        self.iterator_opt(mode, readopts)
     }
 
     pub fn iterator_opt<'a: 'b, 'b>(
         &'a self,
         mode: IteratorMode,
-        readopts: &ReadOptions,
+        readopts: ReadOptions,
```

Comment:
as noted in https://github.com/rust-rocksdb/rust-rocksdb/pull/377#discussion_r403442440, this API pessimization is not necessary